src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6207,7 +6207,9 @@ private BoundExpression MakeMemberAccessValue(BoundExpression expr, DiagnosticBa
                         lookupResult,
                         flags);
 
-                    if (!boundMethodGroup.HasErrors && boundMethodGroup.ResultKind == LookupResultKind.Empty && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
+                    if (!boundMethodGroup.HasErrors && (boundMethodGroup.ResultKind == LookupResultKind.Empty ||
+                        (right is GenericNameSyntax genericNameRight && !IsUnboundTypeAllowed(genericNameRight))) &&
+                        typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
                     {
                         Error(diagnostics, ErrorCode.ERR_BadArity, node, rightName, MessageID.IDS_MethodGroup.Localize(), typeArgumentsSyntax.Count);
```

Comment:
>typeArgumentsSyntax.Count [](start = 122, length = 25)

I see. It looks like the error message isn't really based on any actual method. It is based solely on syntax.